apps/ledger-live-mobile/src/screens/Onboarding/steps/deviceSelection.tsx

```diff
@@ -1,51 +1,170 @@
+            navigateInput: {
+              name: NavigatorName.BaseOnboarding,
+              params: {
+                screen: NavigatorName.SyncOnboarding,
+                params: {
+                  screen: ScreenName.SyncOnboardingCompanion,
+                  params: {
+                    device: null,
+                  },
+                },
+              },
+            },
+            pathToDeviceParam: "params.params.params.device",
+          },
+        },
+      });
+    } else {
+      // TODO: FIX @react-navigation/native using Typescript
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore next-line
+      navigation.navigate(ScreenName.OnboardingUseCase, {
+        deviceModelId,
+      });
+    }
   };
 
   return (
-    <OnboardingView
-      hasBackButton
-      title={t("onboarding.stepSelectDevice.title")}
+    <SafeAreaView
+      style={[{ flex: 1 }, { backgroundColor: colors.background.main }]}
     >
-      <StyledStatusBar barStyle="dark-content" />
-      {devices.map(Device => (
-        <ChoiceCard
-          key={Device.id}
-          event="Onboarding Device - Selection"
-          eventProperties={{ id: Device.id }}
-          testID={`Onboarding Device - Selection|${Device.id}`}
-          onPress={() => next(Device.id)}
-          subTitle={t(`onboarding.stepSelectDevice.${Device.id}`)}
-          subTitleProps={{ variant: "h2", color: "neutral.c100" }}
-          title="Ledger"
-          titleProps={{ variant: "small", color: "neutral.c70" }}
-          Image={<Device fill={colors.neutral.c100} height={80} />}
-        />
-      ))}
+      <Flex
+        px={6}
+        pt={8}
+        mb={7}
+        flexDirection="row"
+        justifyContent="space-between"
+      >
+        <Flex>
+          <TouchableOpacity onPress={handleBack}>
+            <ArrowLeftMedium size={24} />
+          </TouchableOpacity>
+        </Flex>
+        <Flex>
+          <TouchableOpacity onPress={handleHelp}>
+            <HelpMedium size={24} />
+          </TouchableOpacity>
+        </Flex>
+      </Flex>
+      <ScrollListContainer flex={1} px={6} bg="background.main">
+        <Flex flex={1}>
+          <Text variant="h4" mb={3} fontWeight="semiBold">
+            {t("syncOnboarding.deviceSelection.title")}
+          </Text>
+          <Text variant="large" color="neutral.c70" mb={8}>
+            {t("syncOnboarding.deviceSelection.subtitle")}
+          </Text>
+          {devices.map(device => (
+            <DiscoverCard
+              key={device.id}
+              event="Onboarding Device - Selection"
+              eventProperties={{ id: device.id }}
+              testID={`Onboarding Device - Selection|${device.id}`}
+              title={getProductName(device.id)}
+              titleProps={{ variant: "h4", fontSize: 16 }}
+              subTitle={t("syncOnboarding.deviceSelection.brand")}
+              subtitleFirst
+              onPress={() => next(device.id)}
```

Comment:
Possible to use a useCallback for that? Otherwise all cards are re-rendered on every render of this component.

Reply:
as `DiscoverCard` is not a memoized component, it will be re-rendered each time `DeviceSelection` re-renders, even if its props did not change. So it won't change anything to memoized this function